### PR TITLE
BitcoinVault 3keys integration

### PR DESCRIPTION
**Pull request by harzo**

### CODE REVIEW

File: electrumx/server/session.py [hunk 1]
```diff
@@ -1686,3 +1687,35 @@ def truncate_auxpow(self, headers_full_hex, start_height):
...
+class BitcoinVaultElectrumX(ElectrumX):
+
+    async def get_balance(self, hashX):
+        utxos = await self.db.all_utxos(hashX)
+        confirmed = sum(utxo.value for utxo in utxos if utxo.spent_height == 0)
+        incoming = sum(utxo.value for utxo in utxos if utxo.spent_height == -1)
+        outgoing = sum(utxo.value for utxo in utxos if utxo.spent_height > 0)
+        unconfirmed = await self.mempool.balance_delta(hashX)
+        self.bump_cost(1.0 + len(utxos) / 50)
+        return {'confirmed': confirmed, 'unconfirmed': unconfirmed, 'incoming': incoming, 'outgoing': outgoing}
```
Comment (mtwaro) [inline electrumx/server/session.py:1701]: We should use `alert_incoming` and `alert_outcoming`. Only `incoming` and `outgoing` might be misleading.

Comment (harzo) [reply electrumx/server/session.py]: Done

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,27 @@
+aiohttp
```
Comment (kj-crypto) [inline requirements.txt:1]: It is safer to freeze versions in requirements. Maybe by `pip freeze` ?

File: tests/test_vault_transactions.py [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+            '01' + TX_SEGWIT +
+            '01' + AIR_RECOVERY_SEGWIT
+
+        )
+        block = self.coin.block(raw_block, 0)
+        alerts = block.alerts
+
+        self.assertEqual(len(alerts), 1)
+        self.assertIsInstance(alerts[0][0], TxVaultSegWit)
+        self.assertEqual(alerts[0][0].type, VaultTxType.RECOVERY)
```
Comment (kj-crypto) [inline tests/test_vault_transactions.py:179]: new line :)

**Pull request merged**